### ISSUE TITLE
[Helix SDK] Support correlation payloads that are not downloaded to the agent

### ISSUE THREAD

**Issue by safern** [dotnet/arcade#6480]
We're currently driving an effort of making debugging helix crash dumps simpler across dotnet org. There are some caveats here:

1. Some tests don't include debug symbols in the correlation payload or work item payload to save time when downloading assets into the workitem agent. This makes it harder to find matching symbols for the actual binaries as going back to the azure devops build and finding the right artifacts, etc makes it hard for some repos (i.e: dotnet/runtime where around 204 artifacts are published per build). 

2. Some tests do include symbols, like libraries tests. We did this as part of the dumpling effort as symbols were uploaded with the dump so that it was easier to get matching symbols. At the moment, people download helix payloads to debug dumps, which is handy as they already have the symbols... however, this slows down some test runs as we spend a lot of machine time downloading the payloads as they are heavier.

3. With new features like debugging a linux dump in Windows using the cross DAC binaries, people that want to do this, need to get the matching cross DAC for the coreclr that the dump was capture with. We upload this as assets in dotnet/runtime, but again, going back to the pipeline and finding the right pipeline and asset sometimes involves a lot of clicks. 

It would be really nice, if the Helix SDK, supported removing symbols from payloads and uploading all payloads from a correlation payload into a shared correlation symbol container in azure and the workitem symbols into an azure container specific for that workitem, and the correlate that information in the `joblist` file and kusto so that people can use that information to download symbols. This would help a lot to automate things in order to produce a script that downloads the payload, symbols and runs the necessary command to get the debugger or `dotnet-dump` ready with the dump and symbols loaded.

Also, as some of the assets are not included in the payloads like the cross DAC, maybe we can add a feature by metadata to specify we want that to go to a "helper" assets container or even use the same correlation payload symbols container. With something like:

```xml
<HelixCorrelationPayload Include="*">
   <IsSupportPayload>true</IsSupportPayload>
</HelixCorrelationPayload>
```

cc: @markwilkie @Chrisboh @MattGal @ViktorHofer @danmosemsft @davidfowl @hoyosjs

**Comment by markwilkie** [comment]
Adding this to our diagnostic epic (which is part of WF) - which....is a high priority for us.  This doesn't necessarily mean we'll get to this for sure, but it does mean it'll be under serious consideration as we plan for .NET 6.

Also, if this turns out to be an area where other want to partner with us, that is always an option.

**Comment by MattGal** [comment]
I just want to say that we've had times where the Helix client got deeply entangled in corefx/runtime specific functionality, and it bears the vestigial scars of this time yet no longer uses any of that functionality.  This sounds a lot like one of those times. 
 Changes like this need to be made very carefully at the right layer (most likely the Arcade Helix SDK level) if they are to avoid being complex abandoned pieces no longer in use in a few years.

**Comment by markwilkie** [comment]
Agreed - I suspect this would be part of a larger effort to add another layer above what we currently have where we could put common test functionality.  This also implies that to use this, the repos need to test in a similar fashion....

**Comment by safern** [comment]
> I just want to say that we've had times where the Helix client got deeply entangled in corefx/runtime specific functionality

The only piece here that is runtime specific is the cross DAC. We could leave without that last part to include support payloads. However the PDBs/symbols pieces, is not specific to us, I know ASP.NET doesn't include PDBs in their payloads and have to go to the build to get the matching symbols to debug a dump.

cc: @dougbu 

Also I'd like to add, that all the repos would benefit from this as if we don't download the PDBs to the helix agent, dotnet/runtime will use less machine compute times downloading payloads.

**Comment by markwilkie** [comment]
Yes!  Also....there are a number of ways teams are currently using (or not using) Helix so some sort of normalizing would be in order.  But getting to more of a shared place seems like only goodness to me.

**Comment by dougbu** [comment]
> uploading all payloads from a correlation payload into a shared correlation symbol container in azure and the workitem symbols into an azure container specific for that workitem

This sounds complicated to me and I'm not sure it would be a move in the direction @davidfowl is hoping to take our test diagnostic work.
- Why not a _single_ correlation symbol container per queue / job pair❔ Or, perhaps, publish symbols to a unique container directly from the build agent to reduce network hops❔ 
- What about symbols for test applications built on Helix agents e.g. in ASP.NET's template tests❔
- What about symbols coming from other repos e.g. dotnet/runtime symbols for assemblies dotnet/aspnetcore uses❔
- How would this relate to the overall move toward testing in a validation ring against a likely-incoherent SDK❔

On the other hand, I've never hit the specific issue described and defer to @davidfowl 

> I know ASP.NET doesn't include PDBs in their payloads and have to go to the build to get the matching symbols to debug a dump.

FYI we don't use correlation payloads at all.

**Comment by adiaaida** [comment]
Closing because the parent Epic was closed. If you believe this issue should still be worked on, please re-open.
